soccer/gameplay/tactics/wall.py

```diff
@@ -0,0 +1,96 @@
+        self._defense_point = defender_point
+        self.dist_from_mark = dist_from_mark
+        self.defender_spacing = defender_spacing
+
+        self.add_state(Wall.State.defense_wall, 
+                       behavior.Behavior.State.running)
+
+        self.add_transition(behavior.Behavior.State.start,
+                            Wall.State.defense_wall, lambda: True,
+                            "immediately")
+
+        self.add_transition(
+            Wall.State.defense_wall,
+            behavior.Behavior.State.completed, lambda: self.completed(), "robot kicked")
+
+    def on_enter_defense_wall(self):
+        self.remove_all_subbehaviors()
+        midpoint = self.arc_midpoint()
+        angle = (math.atan2(midpoint.x - self.defense_point.x, midpoint.y - self.defense_point.y))
+        for i in range(self.number_of_defenders):
+            pt = None
+            if self.number_of_defenders % 2 != 0:
+                pt = self.point_on_arc(midpoint, angle, i - int(self.number_of_defenders / 2))
+            else:
+                pt = self.point_on_arc(midpoint, angle, i - self.number_of_defenders / 2 + .5)
+            self.add_subbehavior(
+                skills.move.Move(pt),
+                name="robot" + str(i),
+                required=False,
+                priority=self.number_of_defenders - i)
+
+    def completed(self):
+        for bhvr in self.all_subbehaviors():
+             if not bhvr.is_done_running():
+                return False
+        self.remove_all_subbehaviors()
+        return True
+
+    # Finds the point on the wall (arc) at which the defender robot should move to
+    def point_on_arc(self, midpoint, angle, defender_number):
+        x_pt = midpoint.x - defender_number * (constants.Robot.Radius * self.defender_spacing) * math.cos(angle + (defender_number * self.curvature))
+        y_pt = midpoint.y + defender_number * (constants.Robot.Radius * self.defender_spacing) * math.sin(angle + (defender_number * self.curvature))
+        return robocup.Point(x_pt,y_pt)
+
+    # Finds the point on the line between the defense and mark point that is the distance specified from the mark point
+    def arc_midpoint(self):
+        dist_from_defense_and_mark = (self.mark_point - self.defense_point).mag()
+        x_pt = self.mark_point.x + self.dist_from_mark / dist_from_defense_and_mark * (self.defense_point.x - self.mark_point.x)
+        y_pt = self.mark_point.y + self.dist_from_mark / dist_from_defense_and_mark * (self.defense_point.y - self.mark_point.y)
+        return robocup.Point(x_pt, y_pt)
+
+    @property
+    def defense_point(self):
+        return self._defense_point
+
+    @property
+    def mark_point(self):
+        return self._mark_point
+
+    @defense_point.setter
+    def defense_point(self, point):
+        self.remove_all_subbehaviors()
+        self._defense_point = point
+        self.on_enter_defense_wall()
```

Comment:
Calling this feels wrong. I think our general practice has been to create a list of move behaviors in the on_enter or constructor and then have a seperate function that loops through all moves and assigns the updated move to point target. See [here](https://github.com/RoboJackets/robocup-software/blob/master/soccer/gameplay/tactics/line_up.py#L61)